src/net_processing.cpp

```diff
@@ -899,6 +926,14 @@ void PeerManager::UpdateNextReconRequest(std::chrono::microseconds now)
     m_next_recon_request = now + RECON_REQUEST_INTERVAL / m_recon_queue.size();
 }
 
+std::chrono::microseconds PeerManager::NextReconRespond(std::chrono::microseconds now)
+{
+    if (m_next_recon_respond < now) {
+        m_next_recon_respond = now + RECON_RESPONSE_INTERVAL;
```

Comment:
I'm wondering if this might be better as a poisson-timer, but I don't quite understand the implications of all this yet to reach a conclusion.  Intuition: with a fixed delay like this, an adversary with 2 inbound connections can use set reconciliation to measure the set difference between the node's transactions-to-announce in known and arbitrarily precise time slices.

Reply:
I also don't have a good answer, but it's not that simple of an attack: we [have](https://github.com/bitcoin/bitcoin/pull/18261/files#r541017754) Poisson on adding to reconciliation sets.

Having another poison here, well, I don't think it makes anything worse. Maybe you're right.